haystack/nodes/prompt/prompt_node.py

```diff
@@ -232,6 +261,19 @@ def __init__(
         devices: Optional[List[Union[str, torch.device]]] = None,
         **kwargs,
     ):
+        """
+        Creates an instance of HFLocalInvocationLayer used to invoke local Hugging Face models.
```

Comment:
I think it would be helpful to explain the class docstring that we are using the `text2text-generation` HF pipeline in this invocation class. This way we could let users know what model types are supported by linking to the HF docs for the [Text2TextGenerationPipeline](https://huggingface.co/docs/transformers/v4.24.0/en/main_classes/pipelines#transformers.Text2TextGenerationPipeline) and/or link to the [HF models](https://huggingface.co/models?filter=text2text-generation) filtered by the text2textgeneration tag.

Reply:
I wanted to do that @sjrl, but not all HF `text2text-generation` models can support what we are doing here. For now, only T5 flan models have the required capability. In conclusion, I'd rather not add this to docs because then they might start experimenting, and hit roadblocks, report bugs which are not bugs and so on. WDYT?

Reply:
Oh interesting. Why are T5 flan models supported? And in that case, it might be good to state that only T5 flan models are supported in the docstring.

Reply:
Because only T5 flan models have the capability we need. We kinda state that indirectly in the docs that are linked https://docs.haystack.deepset.ai/docs/prompt_node. Do you think we should add that to the py docs themselves? I thought we could keep the list in the haystack docs...either way.

Reply:
No, not input tokens; it is mainly related to the instruction fined tuned training approach used in T5 Flan. Previous models in this generation, like T5 itself, didn't use this approach and consequently didn't "follow" human instructions well. The only capable models are GPT-3 instruct variants trained using the RLHF approach. All other models I tried did not follow instructions well - therefore, we couldn't use them in conjunction with this PromptTemplate approach. You make an excellent point Seb; I'll adjust the docs to reflect this.

Reply:
Added https://github.com/deepset-ai/haystack/pull/3821/commits/994666ef200f0cc5706772d948c678b4759e20ca, but I'll also speak to @agnieszka-m tomorrow to adjust the docs.

Reply:
> No, not input tokens; it is mainly related to the instruction fined tuned training approach used in T5 Flan. Previous models in this generation, like T5 itself, didn't use this approach and consequently didn't "follow" human instructions well. 

Ahh I see! I think this is very relevant information and would be worth adding in the docs. Even a short sentence like `We recommend using LLMs fine-tuned on instruction type datasets otherwise we find that the LLM does not "follow" prompt instructions well. This is why we recommend using T5 flan or OpenAI InstructGPT models.` (or something similar). What do you think? This way users will understand that they can swap in other LLMs that have also been fine-tuned on instruction tasks in the future.

Reply:
Yeah, good point; let me find the right wording for `LLMs fine-tuned on instruction type datasets` that's the critical sentence. Thanks @sjrl

Reply:
`LLMs fine-tuned on collection of datasets phrased as instructions`